File: assignment_seven.cpp
// Copyright (c) 2019 Joseph Palermo All rights reserved.

// Created by: Joseph Palermo
// Created on: December 2019
// This program calculates the volume of an icosahedron

#include <iostream>
#include <iomanip>
#include <array>
#include <numeric>


template<size_t N>
int OddNumber(std::array<int, N> posInt) {

    // input & variables
    int loopCounter;
    int numberFromUser;
    posInt = [];
    positivity [];

    for (loopCounter = 0; loopCounter < (posInt); loopCounter++) {
        std::cout << "Enter a number you want to check: ";
        std::cin >> numberFromUser;
        positivity[loopCounter] = numberFromUser;

    } for (loopCounter = 0; loopCounter < std::array<int, N> positivity); loopCounter++ {
        if positivity[loopCounter] % 2 != 0 {
            oddInt[loopCounter] = positivity([loopCounter])  // help
        } else {
            oddInt[loopCounter] = -1
        }
    }
    return oddInt;
}


int main() {
    // this function calls other functions
    int oddNumber;
    int numberFromUser;
    int posInt;  // should I reference these by array or by list of by integer?
    int check;
    int loopCounter;
    // call function
        std::cout << "This program takes all numbers from 0 to x and outputs "
              "the odd numbers.";
        std::cout << "";
    try {
        std::cout << "Enter amount of numbers you want to check (0-100): ";
        std::cin >> posInt;

        // call function
        numberFromUser = oddNumber(posInt);


        std::accumulate(numberFromUser) / posInt  // help
        std::cin >> check;

        std::cout << "";
        if (check == -1) {
            std::cout << "No odd numbers";
        } else {
            } for (loopCounter = 0; loopCounter < std::array<int, N> numberFromUser) {  // help
                if (numberFromUser[loopCounter]) == -1 {  // help
                    continue;
                } else {
                    std::cout << numberFromUser[loopCounter];  // help
            }
        }
    } catch (std::invalid_argument) {
        std::cout << "";
        std::cout << "Please insert a positive integer";
    }
}

File: assignment_seven.py
#!/usr/bin/env python3

# Created by: Joseph Palermo
# Created on: December 2019
# This program calculates all odd numbers from 0 - x


def odd_number(pos_int):
    # This function calculates the volume of an icosahedron
    counter = 0
    positivity = []
    odd_int = []

    for counter in range(pos_int):
            number_from_user = int(input("Enter a number you want to check: "))
            positivity.append(number_from_user)

    for counter in range(len(positivity)):
        if positivity[counter] % 2 != 0:
            odd_int.append(positivity[counter])
        else: 
            odd_int.append(-1)
    
        
    return odd_int


def main():
    # this function calls other functions

    try:
        print("This program takes all numbers from 0 to x and outputs the "
              "odd numbers.")
        print("")
        pos_int = int(input("Enter amount of numbers you want to check (0-100)"
                            ": "))

        # call function
        number_from_user = odd_number(pos_int)

        check = sum(number_from_user)/pos_int

        print("")
        if check == -1:
            print("No odd numbers")
        else:
            for counter in range (len(number_from_user)):
                if number_from_user[counter] == -1:
                    continue
                else:
                    print(number_from_user[counter])
    except ValueError:
        print("")
        print("Please insert a positive integer")


if __name__ == "__main__":
    main()
